src/RepoAuditor/Plugins/GitHub/Impl/EnableRulesetRequirementImpl.py

```diff
@@ -59,25 +59,58 @@ def _EvaluateImpl(
         query_data: dict[str, Any],
         requirement_args: dict[str, Any],
     ) -> Requirement.EvaluateImplResult:
+        expected_value = self.enabled_by_default
+
         if requirement_args[self.dynamic_arg_name]:
-            # Get the rules for the specified branch
-            rules = query_data.get("rules", [])
+            expected_value = not expected_value
+            provide_rationale = False
+        else:
+            provide_rationale = True
+
+        query_data["__expected_value"] = "Enable" if expected_value else "Disable"
+
+        # Get the rules for the specified branch
+        rules = query_data.get("rules", [])
+
+        # Check if the rule is checked in the ruleset
+        rule_enabled = False
+        ruleset = None
+        for rule in rules:
+            # If the rule is present, it is checked
+            if self.get_configuration_value_func(rule):
+                rule_enabled = True
+                # Get the ruleset associated with the rule
+                ruleset = rule["ruleset"]
+                break
 
-            for rule in rules:
-                if self.get_configuration_value_func(rule):
-                    # Get the ruleset associated with the rule
-                    ruleset = rule["ruleset"]
-                    return self.EvaluateImplResult(
-                        EvaluateResult.Success,
-                        f"Ruleset '{ruleset['name']}' enforces {self.github_settings_value}",
-                    )
+        # If rule is active and we want it enabled,
+        # or rule is inactive and we want it disabled,
+        # return success.
+        if rule_enabled == expected_value:
+            context = (
+                f"{self.github_settings_value} is enabled in Ruleset '{ruleset['name']}'"
+                if expected_value
+                else f"{self.github_settings_value} is disabled"
+            )
+            return self.EvaluateImplResult(
+                EvaluateResult.Success,
+                context,
+            )
 
+        # When rule is active but we want it disabled
+        if rule_enabled and not expected_value:
```

Comment:
I'm having to think about this code quite a bit due to the way that it is currently written. One thing that might help is to rename `expected_value` to `expected_rule_enabled`.

The clause on 89 checks to see if the values are the same (`True` / `True` or `False` / `False`).

The clause on 101 seems to handle the cases (`True` / `False`, although I have to do a lot of thinking to determine that it doesn't handle `True` / `True`, as that is handled above). I'd suggest changing this to `if rule_enabled`.

The clause on 111 seems to handle `False` / `True` but I have to follow all of the logic above to convince myself of this. I'd suggest adding assertions to make this clearer:  
 

```
assert rule_enabled is False
assert expected_value is True
```